consensus/src/state_replication.rs

```diff
@@ -17,7 +17,11 @@ pub trait TxnManager: Send + Sync {
 
     /// Notifies TxnManager about the executed result of the block,
     /// which includes the specifics of what transactions succeeded and failed.
-    async fn notify(&self, block: &Block, compute_result: &StateComputeResult) -> Result<()>;
+    async fn notify(
+        &self,
+        block: &Block,
+        executed_result: Result<&StateComputeResult>,
```

Comment:
why don't we embed the result in the StateComputeResult? this feels a little redundant... will read more ... but figured easier to ask a naive question.

Reply:
Hmm.... @zekun000

Reply:
StateComputeResult represents the successful executed result, I don't understand how do we embed the Result into that? that means we need another struct StateComputeSuccessfulResult to represent the existing struct, don't see what's the benefit.